pkg/controller/master/image/vm_image_controller.go

```diff
@@ -158,32 +150,37 @@ func (h *vmImageHandler) createBackingImage(image *harvesterv1.VirtualMachineIma
 }
 
 func (h *vmImageHandler) createStorageClass(image *harvesterv1.VirtualMachineImage) error {
-	recliamPolicy := corev1.PersistentVolumeReclaimDelete
+	reclaimPolicy := corev1.PersistentVolumeReclaimDelete
 	volumeBindingMode := storagev1.VolumeBindingImmediate
 	sc := &storagev1.StorageClass{
 		ObjectMeta: metav1.ObjectMeta{
-			Name: getImageStorageClassName(image.Name),
+			Name: util.GetImageStorageClassName(image.Name),
 		},
 		Provisioner:          types.LonghornDriverName,
-		ReclaimPolicy:        &recliamPolicy,
+		ReclaimPolicy:        &reclaimPolicy,
 		AllowVolumeExpansion: pointer.BoolPtr(true),
 		VolumeBindingMode:    &volumeBindingMode,
-		Parameters: map[string]string{
-			types.OptionNumberOfReplicas:    "3",
-			types.OptionStaleReplicaTimeout: "30",
-			optionMigratable:                "true",
-			optionBackingImageName:          getBackingImageName(image),
-		},
+		Parameters:           util.GetImageStorageClassParameters(image),
 	}
 
 	_, err := h.storageClasses.Create(sc)
 	return err
 }
 
-func getImageStorageClassName(imageName string) string {
-	return fmt.Sprintf("longhorn-%s", imageName)
+func (h *vmImageHandler) deleteBackingImage(image *harvesterv1.VirtualMachineImage) error {
+	return h.backingImages.Delete(util.LonghornSystemNamespaceName, util.GetBackingImageName(image), &metav1.DeleteOptions{})
 }
 
-func getBackingImageName(image *harvesterv1.VirtualMachineImage) string {
-	return fmt.Sprintf("%s-%s", image.Namespace, image.Name)
+func (h *vmImageHandler) deleteStorageClass(image *harvesterv1.VirtualMachineImage) error {
+	return h.storageClasses.Delete(util.GetImageStorageClassName(image.Name), &metav1.DeleteOptions{})
+}
+
+func (h *vmImageHandler) deleteBackingImageAndStorageClass(image *harvesterv1.VirtualMachineImage) error {
+	if err := h.deleteBackingImage(image); err != nil && !errors.IsNotFound(err) {
+		return err
```

Comment:
could we leave some tips for debugging when deleteBackingImage error?

Reply:
wrangler framework will print the deleteBackingImage error

Reply:
OK, got it!